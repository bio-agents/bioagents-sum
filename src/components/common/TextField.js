import React from 'react'
import { Col, ControlLabel, FormControl, FormGroup, HelpBlock } from 'react-bootstrap'

const TextField = ({ label, help, input }) => (
  <FormGroup controlId={input.name}>
    <Col componentClass={ControlLabel} sm={2}>
      {label}
    </Col>
    <Col sm={10}>
      <FormControl {...input} />
    </Col>
    {help && <HelpBlock>{help}</HelpBlock>}
  </FormGroup>
)

export default TextField
