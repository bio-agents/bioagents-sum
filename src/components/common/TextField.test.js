import React from 'react'
import renderer from 'react-test-renderer'
import TextField from './TextField'

it('Renders TextField correctly', () => {
  const component = renderer.create(
    <TextField label='I want to create my own graph by selecting agents' input={{ value: 'Hello', name: 'snapshot-textfield' }} />
  )

  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
