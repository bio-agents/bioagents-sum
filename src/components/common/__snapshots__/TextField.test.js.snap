exports[`test Renders TextField correctly 1`] = `
<div
  className="form-group">
  <label
    className="col-sm-2 control-label"
    htmlFor="snapshot-textfield">
    I want to create my own graph by selecting agents
  </label>
  <div
    className="col-sm-10">
    <input
      className="form-control"
      id="snapshot-textfield"
      name="snapshot-textfield"
      type={undefined}
      value="Hello" />
  </div>
</div>
`;
